src/Config/ConfigResolver.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Slim\Console\Config;
+
+use InvalidArgumentException;
+use ReflectionClass;
+use Slim\Console\Exception\ConfigNotFoundException;
+
+use function array_filter;
+use function file_exists;
+use function file_get_contents;
+use function implode;
+use function is_array;
+use function is_null;
+use function json_decode;
+use function strpos;
+
+class ConfigResolver
+{
+    public const FORMAT_JSON = 'json';
+    public const FORMAT_PHP = 'php';
+    //public const FORMAT_YAML = 'yaml';
+    private const CONFIG_FILENAME = 'slim-console.config';
+
+    /** @var string|null */
+    private $file = null;
+
+    /** @var string */
+    private $dir;
+
+    public function __construct(string $dir)
+    {
+        $this->dir = $dir;
+    }
+
+    public function loadConfig(): ?Config
```

Comment:
So we can basically instantiate a `ConfigResolver` and then load the config multiple times? Why should we?

Reply:
This resolver should only have one public method called `resolve` which either returns a `Config` object or throws.

Reply:
@adriansuter @l0gicgate 
Should we disallow from loading config again?